src/App.js

```diff
@@ -15,10 +16,41 @@ function App() {
     });
   }
 
+  function editCard() {
+    setCardState({
+      ...cardState,
+      edited: !cardState.edited,
+      checked: (cardState.checked ? !cardState.checked : cardState.checked)
+    });
+  }
+
+  function captionChange(event) {
+    setCardState({
+      ...cardState,
+      caption: event.target.value,
+    });
+  }
+
+  function textChange(event) {
+    setCardState({
+      ...cardState,
+      text: event.target.value
+    });
+  }
```

Comment:
These handlers could be combined by using spread operator and dynamic properties, just pass required property name and get smth like:
```
const changeHandler = (event, property) => {
  changeCardState({ ...cardState, [property] : event.target.value });
};
```
Not so changed at the first glance, however, if there are dozens fields to create handlers for - you'll see the difference.